### ISSUE TITLE
Premature EOF Error at sen2r() Startup

### ISSUE THREAD

**Issue by ryanpmcg** [ranghetti/sen2r#312]
I successfully installed and used several commands from sen2r a couple months ago. Last week I used those same commands and had the following error returned for any function I tried to use:

```
> sen2r()
Error in parse_con(txt, bigint_as_char) : parse error: premature EOF
                                       
                     (right here) ------^
```

I reinstalled all sen2r dependencies: same error.
I reinstalled sen2r and all dependencies: same error.
I installed a clean version of R (4.0.0), sen2r, and all dependencies: same error.

```
> sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.4.0.2 sen2r_1.3.4  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6         lattice_0.20-41      class_7.3-16         shinyWidgets_0.5.1   digest_0.6.25       
 [6] foreach_1.5.0        V8_3.0.2             mime_0.9             R6_2.4.1             leaflet.extras_1.0.0
[11] e1071_1.7-3          httr_1.4.1           geojson_0.3.2        pillar_1.4.3         rlang_0.4.5         
[16] lazyeval_0.2.2       curl_4.3             rstudioapi_0.11      data.table_1.12.8    raster_3.1-5        
[21] geojsonio_0.9.2      shinyjs_1.1          rgdal_1.4-8          jqr_1.1.0            foreign_0.8-78      
[26] htmlwidgets_1.5.1    munsell_0.5.0        compiler_4.0.0       httpuv_1.5.2         pkgconfig_2.0.3     
[31] rgeos_0.5-2          htmltools_0.4.0      mapedit_0.6.0        tibble_3.0.1         httpcode_0.3.0      
[36] codetools_0.2-16     XML_3.99-0.3         crayon_1.3.4         later_1.0.0          sf_0.9-2            
[41] crul_0.9.0           grid_4.0.0           lwgeom_0.2-3         jsonlite_1.6.1       xtable_1.8-4        
[46] lifecycle_0.2.0      DBI_1.1.0            magrittr_1.5         units_0.6-6          scales_1.1.0        
[51] KernSmooth_2.23-16   fs_1.4.1             promises_1.1.0       leaflet_2.0.3        doParallel_1.0.15   
[56] sp_1.4-1             ellipsis_0.3.0       shinyFiles_0.8.0     vctrs_0.2.4          iterators_1.0.12    
[61] tools_4.0.0          crosstalk_1.1.0.1    abind_1.4-5          parallel_4.0.0       fastmap_1.0.1       
[66] RcppTOML_0.1.6       colorspace_1.4-1     stars_0.4-1          maptools_0.9-9       shinydashboard_0.7.1
[71] classInt_0.4-3 

> packageVersion("sen2r")
[1] ‘1.3.4’

> sen2r::load_binpaths()
Error in parse_con(txt, bigint_as_char) : parse error: premature EOF
                                       
                     (right here) ------^
```

**Comment by ranghetti** [comment]
Can you post the output of the following command?
```r
readLines("~/.sen2r/paths.json")
```

**Comment by MiguelAGar** [comment]
Hello, I have exactly the same error. I've been using sen2r for the last weeks, but four days ago this error appeared, and since then, I'm not able to access to the GUI. I've also reinstalled sen2r and a clean version of R, but with the same result.

This is the result for the command:

    > readLines("~/.sen2r/paths.json")
    character(0)

**Comment by ranghetti** [comment]
@MiguelAGar, please report your session info (#186) and attach the file `~/.sen2r/paths.json` mentioned before.

**Comment by MiguelAGar** [comment]
sen2r()
Error in parse_con(txt, bigint_as_char) : parse error: premature EOF
                                       
                     (right here) ------^
> sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_United States.1252 
[2] LC_CTYPE=Spanish_United States.1252   
[3] LC_MONETARY=Spanish_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=Spanish_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sen2r_1.3.4.9001

loaded via a namespace (and not attached):
 [1] httr_1.4.1           shinyFiles_0.8.0     jsonlite_1.6.1      
 [4] foreach_1.5.0        shiny_1.4.0.2        assertthat_0.2.1    
 [7] sp_1.4-1             remotes_2.1.1        pillar_1.4.4        
[10] backports_1.1.6      lattice_0.20-41      glue_1.4.0          
[13] digest_0.6.25        promises_1.1.0       colorspace_1.4-1    
[16] htmltools_0.4.0      httpuv_1.5.2         XML_3.99-0.3        
[19] pkgconfig_2.0.3      httpcode_0.3.0       raster_3.1-5        
[22] stars_0.4-1          xtable_1.8-4         scales_1.1.0        
[25] processx_3.4.2       later_1.0.0          mapedit_0.6.0       
[28] tibble_3.0.1         leaflet.extras_1.0.0 ellipsis_0.3.0      
[31] shinyjs_1.1          withr_2.2.0          geojsonio_0.9.2     
[34] RcppTOML_0.1.6       lazyeval_0.2.2       cli_2.0.2           
[37] magrittr_1.5         crayon_1.3.4         mime_0.9            
[40] maptools_0.9-9       ps_1.3.2             fs_1.4.1            
[43] fansi_0.4.1          doParallel_1.0.15    lwgeom_0.2-3        
[46] foreign_0.8-78       class_7.3-16         pkgbuild_1.0.7      
[49] shinydashboard_0.7.1 tools_4.0.0          data.table_1.12.8   
[52] prettyunits_1.1.1    lifecycle_0.2.0      geojson_0.3.2       
[55] V8_3.0.2             munsell_0.5.0        callr_3.4.3         
[58] compiler_4.0.0       e1071_1.7-3          jqr_1.1.0           
[61] rlang_0.4.5          classInt_0.4-3       units_0.6-6         
[64] grid_4.0.0           iterators_1.0.12     htmlwidgets_1.5.1   
[67] crosstalk_1.1.0.1    shinyWidgets_0.5.1   codetools_0.2-16    
[70] abind_1.4-5          DBI_1.1.0            curl_4.3            
[73] R6_2.4.1             rgdal_1.4-8          fastmap_1.0.1       
[76] rgeos_0.5-2          rprojroot_1.3-2      KernSmooth_2.23-16  
[79] parallel_4.0.0       crul_0.9.0           Rcpp_1.0.4.6        
[82] vctrs_0.2.4          sf_0.9-2             leaflet_2.0.3       
> packageVersion("sen2r")
[1] ‘1.3.4.9001’
> sen2r::load_binpaths()
Error in parse_con(txt, bigint_as_char) : parse error: premature EOF
                                       
                     (right here) ------^
> readLines("~/.sen2r/paths.json")
character(0)

**Comment by ryanpmcg** [comment]
@ranghetti, I get the same result as @MiguelAGar for the readLines command.

**Comment by ranghetti** [comment]
I am trying to reproduce the situation, but I am not encountering this error, neither in your situation (R 4.0, Windows 10). I suspect an encoding problem, but I can not be sure (and I do not know why it was working using previous versions).

@MiguelAGar and @ryanpmcg , can you both make these tries and report them?

1. attach the file `"~/.sen2r/paths.json"` (you can upload it on WeTransfer or similars and provide here the URL) _before_ subsequent operations;
2. try deleting the file  `"~/.sen2r/paths.json"`, restarting and relaunching sen2r;
3. attach again the new file `"~/.sen2r/paths.json"`;
4. try downloading an older version of R (https://cran.r-project.org/bin/windows/base/old/3.6.3/), installing it, setting RStudio to use it (https://support.rstudio.com/hc/en-us/articles/200486138-Changing-R-versions-for-RStudio-desktop ; write me if it is not clear how to do it), restarting RStudio, reinstalling and relaunching sen2r.

Thanks

**Comment by MiguelAGar** [comment]
It seems it was a problem with path.json, because after remove it, the error dissapeared. Then I dowlnloaded again R (4.0.0) and I restarted and relaunched sen2r. After that, I can use the library without any problem.
Anyaway, this is the URL for the first path.json: 
    https://we.tl/t-998rrHJZTh
This is the URL for the second one: 
    https://we.tl/t-pVuRISgdR1

Thank you so much for your help!

**Comment by ranghetti** [comment]
Thanks for the feedback; for unknown reasons, `paths.json` got empty. Since it is not a single isolated issue I will investigate the reason or add a patch to rescan for dependencies in case of empty files.

**Comment by ranghetti** [comment]
Removing GDAL from mandatory dependencies should have fixed that. This issue will be closed when sen2r 1.3.5 will be on CRAN.
